### PR TITLE
:warning: Install the helm chart

### PR DESCRIPTION
**Pull request by tmshort**
# Description

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: internal/controllers/clusterextension_controller.go [hunk 1]
```diff
@@ -347,16 +445,18 @@ func (r *ClusterExtensionReconciler) GenerateExpectedBundleDeployment(o ocv1alph
...
 // SetupWithManager sets up the controller with the Manager.
 func (r *ClusterExtensionReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	err := ctrl.NewControllerManagedBy(mgr).
+	controller, err := ctrl.NewControllerManagedBy(mgr).
 		For(&ocv1alpha1.ClusterExtension{}).
 		Watches(&catalogd.Catalog{},
 			crhandler.EnqueueRequestsFromMapFunc(clusterExtensionRequestsForCatalog(mgr.GetClient(), mgr.GetLogger()))).
 		Owns(&rukpakv1alpha2.BundleDeployment{}).
-		Complete(r)
+		Build(r)
```
Comment (varshaprasad96) [inline internal/controllers/clusterextension_controller.go:453]: We also need to add watches for pods, and all objects that are owned by the Helm release. Can do in followup, just TODO.

File: internal/controllers/common_controller.go [hunk 2]
```diff
@@ -117,3 +117,22 @@ func setDeprecationStatusesUnknown(conditions *[]metav1.Condition, message strin
 		})
 	}
 }
+
+// setInstalledAndHealthyFalse sets the Installed and if the feature gate is enabled, the Healthy conditions to False,
+// and allows to set the Installed condition reason and message.
+func setInstalledAndHealthyFalse(conditions *[]metav1.Condition, message string, generation int64) {
```
Comment (varshaprasad96) [inline internal/controllers/common_controller.go:123]: Nit: Had added status updates in `cluster extension_status.go`. But shouldn't matter.

File: internal/rukpak/helm-operator-plugins/predicate/depedent.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	crtpredicate "sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var log = logf.Log.WithName("predicate")
+
+type GenerationChangedPredicate = crtpredicate.GenerationChangedPredicate
+
+// DependentPredicateFuncs returns functions defined for filtering events
+func DependentPredicateFuncs() crtpredicate.Funcs {
```
Comment (varshaprasad96) [inline internal/rukpak/helm-operator-plugins/predicate/depedent.go:33]: Are we using this predicate anywhere?

**Review by varshaprasad96 — APPROVED** [review]
Just nit, the rest looks good!

File: internal/controllers/common_controller.go (hunk 2 above)
Comment (tmshort) [reply internal/controllers/common_controller.go:123]: There were also a bunch in this file...

File: internal/rukpak/helm-operator-plugins/predicate/depedent.go (hunk 3 above)
Comment (tmshort) [reply internal/rukpak/helm-operator-plugins/predicate/depedent.go:33]: Yes:
https://github.com/operator-framework/operator-controller/pull/762/files#diff-a54f79b762054ccdc31e4011d28cce22af8f54d43e1fddc75745ce042f177f7eR294

**Pull request merged**